eox_nelp/pearson_vue/utils.py

```diff
@@ -26,3 +29,16 @@ def update_xml_with_dict(xml: str, update_dict: dict) -> str:
     xml_dict = xmltodict.parse(xml)
     result = deep_update(xml_dict, update_dict)
     return xmltodict.unparse(result, full_document=False)
+
+
+def generate_client_authorization_id(user_id: int, course_id: str) -> str:
+    """Use course_enrollment_id to generate client_authorization_id for EAD requests.
+    Args:
+        user_id (int): user id
+        course_id (str): course id
+    Returns:
+        str: string that represents the client_auth_id
+    """
+    course_enrollment = CourseEnrollment.objects.get(user_id=user_id, course_id=course_id)
```

Comment:
This is a simple solution and I like that however this strongly depends on the course enrollment id and that makes me wonder what will happen if one day we need to delete  a course enrollment record to perform a support operation, if we create the enrollment again the  id will be different and we won't be able to track previous records

Reply:
You have a good point, if the course-enrollment is deleted we could not track it. 
One solution is to not make hard deletes of course enrollments and only use the `is_active` flag.
![image](https://github.com/eduNEXT/eox-nelp/assets/51926076/9dbdcad8-66a4-4c80-a109-89bc8871a3cb)
If not I could create an id with
```python

client_authorization_id =  f"{user_id}-{org}+{course}+{run}"
```
something like :
![image](https://github.com/eduNEXT/eox-nelp/assets/51926076/bf593a54-e8c2-4c69-b558-41e931e8fecd)